crates/autopilot/src/run_loop.rs

```diff
@@ -89,7 +90,43 @@ impl RunLoop {
 
     async fn single_run_(&self, id: AuctionId, auction: &Auction) {
         tracing::info!("solving");
-        let mut solutions = self.solve(auction, id).await;
+        let solutions = self.solve(auction, id).await;
+
+        // Validate solutions and filter out invalid ones.
+        let mut solutions = solutions
+            .into_iter()
+            .filter_map(|(index, response)| {
+                if response.solutions.is_empty() {
+                    tracing::debug!(driver = ?self.drivers[index].url, "driver sent zero solutions");
+                    return None;
+                }
+
+                Some(
+                    response
+                        .solutions
+                        .into_iter()
+                        .filter_map(|solution| {
+                            if solution.score == U256::zero() {
```

Comment:
Do we really want to `warn` in this case? I believe today a lot of solvers send us trivial solutions when they don't find anything. Do we need the added verbosity here?

Reply:
Right now, I would expect `driver` to [filter out empty solutions](https://github.com/cowprotocol/services/blob/main/crates/driver/src/domain/competition/mod.rs#L72) and don't send them to `autopilot`, but I guess you are right since we don't have this guarantee anywhere in the driver interface. Then it probably should be `debug`.

Reply:
Updated.